extensions/interactions/NumberWithUnits/directives/NumberWithUnits.js

```diff
@@ -135,9 +136,40 @@ oppia.directive('oppiaShortResponseNumberWithUnits', [
   }
 ]);
 
-// Rule evaluation for number with units (will be implemented in M1.3).
+// Rule evaluation for number with units.
 oppia.factory('numberWithUnitsRulesService', [
   'NumberWithUnitsObjectFactory', 'FractionObjectFactory',
   function(NumberWithUnitsObjectFactory, FractionObjectFactory) {
+    var isEqualValues = function(answer, inputs) {
+      if (answer.type === 'fraction') {
+        return answer.fraction.toString() === inputs.fraction.toString();
+      } else {
+        return answer.real === inputs.real;
+      }
+    };
+
+    return {
+      IsEqualTo: function(answer, inputs) {
+        answer = NumberWithUnitsObjectFactory.fromDict(answer);
```

Comment:
Why not do a simple check: `angular.equals(answer, inputs.f)`? Wouldn't that work too?

Reply:
No because we need to convert units in dict format first else they'll never be same even with slightest difference. That's why, I have to check these conditions separately.

Reply:
Unit is stored in dict format in the backend. So I dont see what is the issue. The incoming arguments are dict so why is unit not a dict?

Reply:
I mean in the incoming object dict, unit is stored as unit object string and not as a dict. Using `toDict()` it can be converted in dict format. So, I needed to convert it to dict first and then, checked the conditions separately.

Reply:
Can you explain to me why incoming dict has units stored in string? The incoming object should have exactly same schema as the one defined in `objects.py` file.